client/src/document/hooks.ts

```diff
@@ -75,27 +79,24 @@ function prevHeading(heading: Element) {
 function addBreakoutButton(
   element: Element | null,
   id: string,
-  doc: Doc,
   code: EditorContent,
   locale: string
 ) {
   if (!element || element.querySelector(".play-button")) return;
-  const button = document.createElement("button");
-
-  button.textContent = "Play";
-
-  button.setAttribute("type", "button");
-  button.setAttribute("class", "play-button external");
-  button.title = "Open in Playground";
-  element.appendChild(button);
-
-  button.onclick = async () => {
-    const key = `play-${id}-${doc.mdn_url}`;
-    sessionStorage.setItem(key, JSON.stringify(code));
-    const url = new URL(window?.location.href);
-    url.pathname = `/${locale}/play`;
-    url.searchParams.set("sample", key);
-    window.location.href = url.href;
+  const a = document.createElement("a");
+
+  a.textContent = "Play";
+
+  a.setAttribute("class", "play-button external");
+  a.href = `/${locale}/play`;
+  a.setAttribute("data-play", id);
+  a.title = "Open in Playground";
+  element.appendChild(a);
+
+  a.onclick = async (e) => {
```

Comment:
A not very nice side effect of this that modifier keys are ignored and the playground opens in the same tab, and middle clicking (as well as right-click-context "open in new tab" or "open in new window") just open a new tab/window with an empty playground.

From my understanding of `sessionStorage` it *will* get duplicated to links opened in new tabs/windows, so I think the logic here needs to be to allow default, not set the `window.location`, but to set the sessionStorage before the default action takes place.

Reply:
Button it is. 👍